src/ProjectAudioManager.cpp

```diff
@@ -1089,10 +1076,18 @@ void ProjectAudioManager::OnCommitRecording()
 
 void ProjectAudioManager::OnSoundActivationThreshold()
 {
-   auto &project = mProject;
    auto gAudioIO = AudioIO::Get();
-   if ( gAudioIO && &project == gAudioIO->GetOwningProject().get() ) {
-      wxTheApp->CallAfter( [this]{ Pause(); } );
+   if (gAudioIO)
+   {
+      bool canStop =  CanStopAudioStream();
+
+      gAudioIO->SetPaused(!gAudioIO->IsPaused());
+
+      if (canStop)
+      {
+         // this is what ::OnPause really boils down to, when we consider that we are not scrubbing
+         SetPaused( !Paused() );
```

Comment:
This calls SetPaused on ProjectAudioManager.  It is only changing the boolean `mPaused`.

Strictly speaking, this means the boolean ought to be an atomic, tested and updated with relaxed memory ordering, because it is the non-main thread changing it here, but it is tested elsewhere in the main thread.

Reply:
Completely right, I did not see this potential problem! I will do as you advice.

Reply:
I made the mPaused member an atomic, but in order to be able to toggle it atomically, I had to make it an int; please have a look at my latest commit and let me know what do you think.